src/bigqueryClient.ts

```diff
@@ -0,0 +1,46 @@
+import * as vscode from 'vscode';
+import { BigQuery } from '@google-cloud/bigquery';
+
+let bigquery: BigQuery | undefined;
+let authenticationCheckInterval: NodeJS.Timeout | undefined;
+
+// Function to create or recreate the BigQuery client
+export async function createBigQueryClient() {
+    try {
+        bigquery = new BigQuery();
+        vscode.window.showInformationMessage('BigQuery client created successfully.');
+    } catch (error) {
+        vscode.window.showErrorMessage(`Error creating BigQuery client: ${error}`);
+    }
+}
+
+// Function to check authentication and recreate the client if necessary
+export async function checkAuthentication() {
+    if (!bigquery) {
+        await createBigQueryClient();
+        return;
+    }
+
+    try {
+        // Perform a simple query to check authentication status
+        await bigquery.query('SELECT 1');
```

Comment:
this will get called 3 times on every single dry run. We perhaps don't want that. Would it be possible to only trigger this check on a specific error type reported in the issue and then trigger re-authentication.